cmd/secretcli/secret20.go

```diff
@@ -0,0 +1,371 @@
+		Args: cobra.MinimumNArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+
+			inBuf := bufio.NewReader(cmd.InOrStdin())
+			cliCtx := context.NewCLIContextWithInput(inBuf).WithCodec(cdc)
+
+			contractAddrStr, err := addressFromBechOrLabel(args[0], cliCtx)
+			if err != nil {
+				return err
+			}
+
+			contractAddr, err := sdk.AccAddressFromBech32(contractAddrStr)
+			if err != nil {
+				return errors.New("invalid contract address or label")
+			}
+
+			var src = rand.New(rand.NewSource(time.Now().UnixNano()))
+
+			byteArr := make([]byte, 32) // can be simplified to n/2 if n is always even
+			if _, err := src.Read(byteArr); err != nil {
+				panic(err)
+			}
+
+			randomData := hex.EncodeToString(byteArr)[:64]
+
+			msg := createViewingKeyMsg(randomData)
+
+			return cli.ExecuteWithData(cmd, contractAddr, msg, "", false, "", "", cdc)
+		},
+	}
+
+	return cmd
+}
+
+func s20SetViewingKey(cdc *codec.Codec) *cobra.Command {
+
+	cmd := &cobra.Command{
+		Use:   "set-viewing-key [contract address or label]",
+		Short: "sets the viewing key for your account",
+		Long: `This command is useful if you want to manage multiple secret tokens with the same viewing key. WARNING:
+This should only be used to duplicate keys created with the create-viewing-key command, or if you really really know what
+you're doing`,
+		Args: cobra.MinimumNArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+
+			inBuf := bufio.NewReader(cmd.InOrStdin())
+			cliCtx := context.NewCLIContextWithInput(inBuf).WithCodec(cdc)
+
+			contractAddrStr, err := addressFromBechOrLabel(args[0], cliCtx)
+			if err != nil {
+				return err
+			}
+
+			contractAddr, err := sdk.AccAddressFromBech32(contractAddrStr)
+			if err != nil {
+				return errors.New("invalid contract address or label")
+			}
+
+			msg := setViewingKeyMsg(args[1])
+
+			return cli.ExecuteWithData(cmd, contractAddr, msg, "", false, "", "", cdc)
+		},
+	}
+
+	return cmd
+}
+
+func s20DepositCmd(cdc *codec.Codec) *cobra.Command {
+
+	cmd := &cobra.Command{
+		Use:   "deposit [contract address or label]",
+		Short: "Convert your SCRT into a secret token",
+		Long:  `Convert your SCRT into a secret token. This command will only work if the token supports native currency conversion`,
+		Args:  cobra.MinimumNArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+
+			inBuf := bufio.NewReader(cmd.InOrStdin())
+			cliCtx := context.NewCLIContextWithInput(inBuf).WithCodec(cdc)
+
+			contractAddrStr, err := addressFromBechOrLabel(args[0], cliCtx)
+			if err != nil {
+				return err
+			}
+
+			contractAddr, err := sdk.AccAddressFromBech32(contractAddrStr)
+			if err != nil {
+				return errors.New("invalid contract address or label")
+			}
+
+			msg := depositMsg()
+
+			amountStr := viper.GetString(flagAmount)
+
+			return cli.ExecuteWithData(cmd, contractAddr, msg, amountStr, false, "", "", cdc)
+		},
+	}
+	cmd.Flags().String(flagAmount, "", "")
+	return cmd
+}
```

Comment:
why did you decide to pass the amount to `deposit` as a flag, rather than as a positional argument, like in `withdraw`?

Reply:
because you actually need to send native coins

Reply:
Ah, so we want the commands to look more like the native commands?

Reply:
A deposit is literally "send SCRT and get sSCRT", so you have to have it actually work:)